flask_user/forms.py

```diff
@@ -210,7 +221,7 @@ def validate(self):
             user, user_email = user_manager.find_user_by_email(self.email.data)
 
         # Handle successful authentication
-        if user and user.password and user_manager.verify_password(self.password.data, user):
+        if user and user_manager.verify_password(self.password.data, user):
```

Comment:
This minimizes the assumption that `password` is a field on the user entity. In reality, `verify_password` should *never* be returning `True` if the user has no password at all.